ext/serializer.c

```diff
@@ -1152,10 +1152,77 @@ static void dd_set_entrypoint_root_span_props_end(zend_array *meta, int status,
         ZVAL_STR(&status_zv, status_str);
         zend_hash_str_update(meta, ZEND_STRL("http.status_code"), &status_zv);
 
-        if (status >= 500 && !ignore_error) {
-            zval zv = {0}, *value;
-            if ((value = zend_hash_str_add(meta, ZEND_STRL("error.type"), &zv))) {
-                ZVAL_STR(value, zend_string_init(ZEND_STRL("Internal Server Error"), 0));
+        // Only check status codes if not ignoring errors
+        if (!ignore_error) {
+            bool is_error = false;
+            bool is_custom_error = false;
+
+            // Get server error configuration
+            zend_array *cfg = get_DD_TRACE_HTTP_SERVER_ERROR_STATUSES();
+
+            // If configuration exists and has entries, check against it
+            if (cfg && zend_hash_num_elements(cfg) > 0) {
+                // Check custom configuration first
+                zend_string *str_key;
+                ZEND_HASH_FOREACH_STR_KEY(cfg, str_key) {
+                    if (str_key) {
+                        const char *s = ZSTR_VAL(str_key);
+                        char *dash = strchr(s, '-');
+
+                        if (dash) {
+                            // Range like "500-599"
+                            int start, end;
+                            if (sscanf(s, "%d-%d", &start, &end) == 2) {
+                                if (status >= start && status <= end) {
+                                    is_error = true;
+                                    is_custom_error = true;
+                                    break;
+                                }
+                            }
+                        } else {
+                            // Single status code
+                            int code = atoi(s);
```

Comment:
```suggestion
                            zend_long code = ZEND_ATOL(s);
```
We use the generic version from zend here